TinkerWorX.SharpCraft.Launcher/Program.cs

```diff
@@ -19,10 +19,29 @@ public static void Main(String[] args)
             // This fixes the current directory in case it is run from outside the launcher path.
             Environment.CurrentDirectory = Path.GetDirectoryName(typeof(SharpCraftApplication).Assembly.Location);
 
+            // Check for optional map specified to launch
+            string mapFile = ""; // Is there a loadfile arg specified?
+            for (var i = 0; i < args.Length; i++)
+            {
+                if (args[i].Contains(".w3x"))
```

Comment:
This needs to also take non expansion maps (w3m) and replays (w3g) into account. Alternatively, you could look for the "-loadfile" argument, since the next argument will be the file. This way it doesn't rely on file extensions which can change for any reason.